### PR TITLE
Order utilities clockwise where applicable

### PR DESCRIPTION
**Pull request by thecrypticace**
Fixes https://github.com/tailwindlabs/prettier-plugin-tailwindcss/issues/147

We're gonna change things so the appropriate `inset`, `padding`, `margin`, etc… utilities are in a clockwise sort order.

### CODE REVIEW

File: src/corePlugins.js [hunk 1]
```diff
@@ -631,14 +631,12 @@ export let corePlugins = {
...
-        ['right', ['right']],
-        ['bottom', ['bottom']],
-        ['left', ['left']],
-      ],
+      [['top', ['top']]],
+      [['right', ['right']]],
+      [['end', ['inset-inline-end']]],
+      [['bottom', ['bottom']]],
+      [['left', ['left']]],
+      [['start', ['inset-inline-start']]],
```
Comment (RobinMalfait) [inline src/corePlugins.js:639]: Do you think we should keep `start` and `end` as-is? This change would sort it `end` first then `start` which seems odd? 🤔

Comment (thecrypticace) [reply src/corePlugins.js:639]: Good question. Yo @adamwathan what do you think?

Comment (adamwathan) [reply src/corePlugins.js:639]: Yeah I put them there on purpose but doing some more research, it looks like the logical ones will be flipped when/if that’s ever supported:

image

So we should go top, right, inline-start, bottom, left, inline-end I think.

**Comment by adamwathan** [comment]
Kinda second guessing some of this a bit, I think I would hate for `pr-*` to be sorted before `pl-*`. I think we should have an opinion about this in Tailwind but not convinced it should be TRBL 👀

**Comment by adamwathan** [comment]
Kinda second guessing some of this a bit, I think I would hate for `pr-*` to be sorted before `pl-*`. I think we should have an opinion about this in Tailwind but not convinced it should be TRBL 👀

**Comment by teddybradford** [comment]
A different approach could be to sort from
- inline-axis to block-axis
- inline-start to inline-end
- block-start to block-end
and
- x-axis to y-axis
- left to right
- top to bottom

But I think this sorting might be biased for Western locales?

**Comment by adamwathan** [comment]
The people have spoken! Let's do:

- top
- bottom
- inline-start
- left
- inline-end
- right

image

**Comment by Wolfgrammatikalazeredr** [comment]
Mo humanistic

**Comment by Wolfgrammatikalazeredr** [comment]
Wowza

**Review by Wolfgrammatikalazeredr — APPROVED** [review]

**Comment by Wolfgrammatikalazeredr** [comment]
Somwhere stream on through.

**Comment by Wolfgrammatikalazeredr** [comment]
Anyone?

**Comment by kachkaev** [comment]
@thecrypticac what is the current status of this PR? Happy to help with testing, if needed!

**Comment by thecrypticace** [comment]
I'll re-open this PR once we've worked through the decisions we want to make here.

**Pull request closed without merging**